File: bench/string_ops/StringOps.java
// 10000回の upper-case + 結合
public class StringOps {
    public static void main(String[] args) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < 10000; i++) {
            result.append(("item-" + i).toUpperCase());
        }
        System.out.println(result.length());
    }
}

File: bench/string_ops/string_ops.clj
;; string-ops: 文字列操作
;; N回の文字列結合 + 変換
(require '[clojure.string :as str])
(println
 (count
  (reduce str
          (map #(str/upper-case (str "item-" %))
               (range 10000)))))
